Frontend/MoptPaymentPayone/Views/frontend/checkout/ajax_cart_amazon.tpl

```diff
@@ -9,7 +9,7 @@
     {include file="frontend/checkout/script-amazonpay.tpl"}
 
     <script>
-    {if $smarty.server.REQUEST_SCHEME === 'https'}
+    {if $smarty.server.REQUEST_SCHEME === 'https' || $smarty.server.HTTPS === 'on' || $smarty.server.HTTP_HTTPS === 'on' || $smarty.server.HTTP_X_FORWARDED_PROTO === 'https'}
```

Comment:
This should be done using a new variable with value of $request->isSecure().
Don't implement such things by own. Shopware already uses trusted proxies from Symfony. https://symfony.com/doc/current/deployment/proxies.html

Reply:
Thanks for the tip! Does this work as far back as Shopware 5.2 (which is the oldest version we offically support)?

Reply:
Yes since Shopware 4

Reply:
nice!